crates/bevy_ecs/src/change_detection.rs

```diff
@@ -131,6 +131,11 @@ pub trait DetectChangesMut: DetectChanges {
     /// This is useful to ensure change detection is only triggered when the underlying value
     /// changes, instead of every time it is mutably accessed.
     ///
+    /// If you're dealing with non-trivial structs which have multiple fields of non-trivial size,
+    /// then consider applying a [`map_unchanged`](Mut::map_unchanged) beforehand to prevent the
```

Comment:
Linking to only `Mut::map_unchanged` is not totally correct, as this situation can happen with `MutUntyped`, `ResMut` and `NonSendMut` too. However I don't see a way to fix this without verbosely listing all of them, so ideas are welcome. If nobody finds a better way to express this I would keep it like this.

Reply:
I think this is adequately clear: users who are at this level of sophistication will be able to piece this together.

Reply:
I hope I handled this adequately to make everyone here happy